### ISSUE TITLE
Old version of Etherpad

### ISSUE THREAD

**Issue by JohnMcLear** [sandstorm-io/sandstorm#235]
I'm not sure where I should report this but the version of Etherpad you guys are supporting is really old and some minor security issues such as path leaks that have been fixed in later stable releases. Can you please update your port?

**Comment by ocdtrekkie** [comment]
I'd also love for the Sandstorm Etherpad port to get an update, but I did want to mention that it's very unlikely any security issues present in Etherpad would be functional in the Sandstorm version.

**Comment by JohnMcLear** [comment]
As the author of the security issues and the patches that fix them, I beg to differ...

**Comment by ocdtrekkie** [comment]
By design, Sandstorm does a LOT of things to ensure individual documents have no ability to affect anything outside of themselves.

If there's such a vulnerability that does affect Sandstorm, the core devs (I'm not one) would probably be more than happy to hear about it, as they are doing a lot to ensure that every document is entirely isolated from the outside world.

**Comment by JohnMcLear** [comment]
Yea no offence but you guys just need to update.  Having a discussion about the security vulns here doesn't help anyone.

RE "every document is isolated from the outside world", that is a completely false statement and should be revised.  All Sandstorm does is build off our trust model, if our trust model is broken (which it was) then it doesn't matter what Sandstorm does..

Hope that helps clear things up @ocdtrekkie

**Comment by ocdtrekkie** [comment]
It really doesn't. Every Sandstorm document is a completely isolated process that is unable to talk to any others. Every document is it's own isolated running copy of the application. Sandstorm manages all of the security for access to that application.

**Comment by JohnMcLear** [comment]
*sighs.  Sorry but I'm done with this.  My point is still valid, I know what I'm talking about here.  You need to update.

**Comment by ocdtrekkie** [comment]
"I know what I'm talking about here" - I'm sure you're very versed in Etherpad, but probably not so much in Sandstorm.

I'm not a Sandstorm.io employee/developer. And yes, I'd like to see the app updated.

**Comment by paulproteus** [comment]
Hi @JohnMcLear -- thanks for filing this bug. We'll work on updating it.

Some meta-remarks:

@JohnMcLear -- I want to make sure people like you, who are maintainers of projects with Sandstorm ports, are as happy with Sandstorm as possible. It's a goal of Sandstorm is to bring more users to great open source web apps like Etherpad. We'd even love it if upstream maintenance communities become interested in maintaining the Sandstorm port.

By contrast, in reading this conversation, when I see things like "I'm done with this", I sense frustration, and I want you to have a better experience with our community than that.

@ocdtrekkie -- thanks for chiming in about Sandstorm's extra security features. I think it's great to say that once. To honor all the work that @JohnMcLear does on Etherpad, and to make sure all the GitHub notifications they get via the Sandstorm community are of high value, I suggest trying to avoid low-information back-and-forths about security. To help make sure future conversations provide more information, I filed https://github.com/sandstorm-io/sandstorm/issues/236 so that we can have a wiki page to send upstream project maintainers, in case they want to use that to determine if Sandstorm's platform security features do indeed defang an attack.

Having said all that, I totally sympathize with the desire to talk about how cool Sandstorm is. I recently engaged in what I might now describe as low-information back-and-forths about security right here -- https://news.ycombinator.com/item?id=8972066 -- so I totally get that you're excited and want to help people know that Sandstorm is eye-openingly different. (-:

@JohnMcLear , thanks for filing this issue.
